spring-cloud-commons/src/main/java/org/springframework/cloud/commons/publisher/FluxFirstNonEmptyEmitting.java

```diff
@@ -277,23 +293,26 @@ int resignFromRace() {
 		@Nullable
 		public Object scanUnsafe(Attr key) {
 			if (key == Attr.PARENT) {
-				return s;
+				return ReflectionUtils.getField(s, this);
 			}
 			if (key == Attr.CANCELLED) {
 				return parent.cancelled;
 			}
 
-			return InnerOperator.super.scanUnsafe(key);
+			return doScanUnsafe(key);
 		}
 
-		@Override
-		public void onSubscribe(Subscription s) {
-			set(s);
+		@Nullable
+		Object doScanUnsafe(Attr key) {
```

Comment:
as commented above, this can be melded back into `scanUnsafe` to get:

```
@Nullable
public Object scanUnsafe(Attr key) {
    if (key == Attr.ACTUAL) return actual;
    if (key == Attr.CANCELLED) return parent.cancelled;
    return super.scanUnsafe(key);
}
```